network/basics/uri.py

```diff
@@ -49,6 +48,12 @@
       - path of where to download the file to (if desired). If I(dest) is a directory, the basename of the file on the remote server will be used.
     required: false
     default: null
+  source:
+    description:
+      - path to binary file to be uploaded to the endpoint
+    required: false
+    default: null
+    version_added: '1.9.7'
```

Comment:
This version should be the next major release version, which at this point should be likely `2.3`